### PR TITLE
Searches mongodb collection for search query

### PR DESCRIPTION
**Pull request by merezal**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -127,6 +133,15 @@ async function lookupQuiz(quizID, completion) {
         completion(result[0])
     });
 }
+//description tags name
+async function searchCollection(searchTerm, completion) {
+    var query = searchTerm.split('+').join(' ');
+    database.collection(quizCollection).find({$or: [{name: {$regex: query}},{description: {$regex: query}},{tags: {$in: [query]}}]}).toArray(function(err, result) {
```
Comment (julianfortune) [inline server.js]: Can you use the mongobd search functionality please? This looks very brittle to me. https://docs.mongodb.com/manual/text-search/

File: public/search.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 var search_box = document.querySelector('#nav_search_box');
 search_box.addEventListener('change', search);
 function search(event){
-    var url='/search/'+search_box.value.replace(/\s+/g, '').toLowerCase();
+    var url='/search/'+search_box.value.replace(/\s+/g, '+');
```
Comment (julianfortune) [inline public/search.js]: I like the plusses. Why not to lowercase? Just no need? Do you think the link would look better with lowercase?

File: public/style.css [hunk 3]
```diff
@@ -138,10 +138,10 @@ input.nav_search_box{
 .num_count {
   font-family: var(--quiz-title-font);
   font-size: 20px;
-  position: absolute;
+  float: right;
```
Comment (julianfortune) [inline public/style.css]: 👌🏼

File: public/style.css [hunk 4]
```diff
@@ -138,10 +138,10 @@ input.nav_search_box{
...
   top: 0px;
   right: 50px;
-  color: var(--darkgrey);
+  color: var(--light-text);
```
Comment (julianfortune) [inline public/style.css]: Can you split the `preview` css for `.num_count` into a different class? This change will break the quiz page.

**Review by julianfortune — DISMISSED** [review]
Couple of things 🧐

File: public/search.js (hunk 2 above)
Comment (merezal) [reply public/search.js]: I felt like case-sensitive searches could be useful, but if not I can just tack that back on.

**Comment by julianfortune** [comment]
Did a little testing, and we should make it so if you only type in spaces it won't search, and I think it should be case insensitive

File: server.js [hunk 5]
```diff
@@ -46,13 +46,19 @@ app.get('/edit/:quizID', function(req, res, next) {
...
+
+    searchCollection(searchTerm, function(quizzes){
+        results=quizzes
+        console.log("Results:", quizzes)
+        res.status(200).render('results', {
+            title: 'Search Results',
+            search_results: results
+        })
+    })
     // Replace with database function ^^^
```
Comment (julianfortune) [inline server.js]: Kill this please

File: server.js [hunk 6]
```diff
@@ -46,13 +46,19 @@ app.get('/edit/:quizID', function(req, res, next) {
 app.get('/search/:searchTerm', function(req, res, next) {
     var searchTerm = req.params.searchTerm
 
-    var results = require("./exampleQuizzes.json")
+    var results={}// = require("./exampleQuizzes.json")
```
Comment (julianfortune) [inline server.js]: The commented code can be removed

File: server.js [hunk 7]
```diff
@@ -46,13 +46,19 @@ app.get('/edit/:quizID', function(req, res, next) {
...
+
+    searchCollection(searchTerm, function(quizzes){
+        results=quizzes
```
Comment (julianfortune) [inline server.js]: You can just rename `quizzes` to `results` in the` function()`

**Comment by julianfortune** [comment]
Also, can you add a piece of text to the top of the top above the results that says like `Results for "Testing"` and if there are no results can you add some text that says `No results found for "Testing".`

**Review by thuyvyng — APPROVED** [review]

File: server.js (hunk 5 above)
Comment (merezal) [reply server.js]: What do you mean here?

File: public/search.js (hunk 2 above)
Comment (merezal) [reply public/search.js]: nvm its not case-sensitive because of db index searching now

**Comment by julianfortune** [comment]
Changes look amazing! 🍾 Don’t we need to create a text index to search though? MongoDB says `To perform text search queries, you must have a text index on your collection.` https://docs.mongodb.com/manual/core/index-text/

**Comment by merezal** [comment]
> Changes look amazing! champagne Don’t we need to create a text index to search though? MongoDB says `To perform text search queries, you must have a text index on your collection.` https://docs.mongodb.com/manual/core/index-text/

There is a text index on the mongodb server

**Comment by merezal** [comment]
I went ahead and just did the recent quizzes in this branch because I needed my partials.

**Review by thuyvyng — DISMISSED** [review]
make the featured quizzes actually links

**Review by julianfortune — APPROVED** [review]
This looks *very nice* 👌🏼👌🏼

**Pull request merged**